src/ai/PathFinder.cpp

```diff
@@ -151,29 +151,25 @@ class PathFinder::OpenNodeList {
 
 class PathFinder::ClosedNodeList {
 	
-	typedef std::vector<Node*> NodeList;
-	NodeList nodes;
+	typedef std::vector<bool> NodeList;
+	NodeList *nodes;
 	
 public:
+
+	ClosedNodeList(int nodeCount) {
+		nodes = new NodeList(nodeCount, false);
+	}
 	
 	~ClosedNodeList() {
-		for(NodeList::iterator i = nodes.begin(); i != nodes.end(); ++i) {
-			delete *i;
-		}
+		delete nodes;
 	}
 	
 	void add(Node * node) {
-		nodes.push_back(node);
```

Comment:
This previously transferred ownership of the Node object to the ClosedNodeList, which was then responsible for deleting it at the end. With your new code, the Node objects will never be deleted.